core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMaterializedViewDefinition.java

```diff
@@ -80,6 +80,10 @@ public ConnectorMaterializedViewDefinition(
         if (columns.isEmpty()) {
             throw new IllegalArgumentException("columns list is empty");
         }
+
+        if (owner.isEmpty()) {
+            throw new IllegalArgumentException("owner must be present");
```

Comment:
what if we change the `@JsonCreator` ctor as well?
i think jackson-produced message would be good as well

Reply:
> what if we change the @JsonCreator ctor as well?

I didn't want to break backward compatibility of already serialized materialized view definitions.

Reply:
you break it anyway -- if the serialized value doesn't have owner, it will fail during deserialization. It doesn't make a big difference whether it fails in jackson code vs in the constructor validtion as long as the raised message is informative.

Reply:
It wouldn't fail if the serialized value does have owner. But if I change jackson constructor it would then fail then too. Technically, someone could already serialized such MV in metastore (we do something like that for `ConnectorViewDefinition` in Hive) and this change would break valid MV then.

Reply:
> It wouldn't fail if the serialized value does have owner. But if I change jackson constructor it would then fail then too.

I don't think this is the case (`com.fasterxml.jackson.datatype.jdk8.Jdk8Module` makes `Optional<String>` and `String` serialize the same way)

Reply:
I'm pretty sure it's not the case. See https://github.com/trinodb/trino/pull/7318#discussion_r596967618. Optional gets serialized into nested structure

Reply:
We don't set `com.fasterxml.jackson.datatype.jdk8.Jdk8Module#configureAbsentsAsNulls` to `true` (default is `false`)

Reply:
That doesn't align with my memory, but i'd need to check this out.

If so, we should definitely have a test exercising the persisted form build with a previous version against current code.
Especially if serialized form is an effect of interaction between SPI class (types, annotations) and a connector setup of `ObjectMapper`.

Reply:
Ok It might work. Experiment:
```
        ConnectorViewDefinition view = new ConnectorViewDefinition("foo", Optional.of("bar"), Optional.empty(), ImmutableList.of(new ConnectorViewDefinition.ViewColumn("f", TypeId.of("foo"))), Optional.empty(), Optional.empty(), false);
        String s = VIEW_CODEC.toJson(view);
```
```
{"originalSql":"foo","catalog":"bar","columns":[{"name":"f","type":"foo"}],"runAsInvoker":false}
```